gdrive.py

```diff
@@ -1,571 +1,18 @@
-        self.__operation = operation
-        self.__start_time = time.time()
-        self.__worker = threading.Thread(target=self.__work, daemon=True)
-        self.__worker.start()
-
-    def send(self, status):
-        """Try to send a new value to the channel."""
-        self.__channel.send(status)
-
-    def close(self):
-        """Close the channel and join the thread."""
-        self.__channel.close()
-        self.__worker.join()
-
-    def __work(self):
-        """Print logs while the channel is open and receiving values."""
-        last_size = 0.0
-        last_time = self.__start_time
-        while self.__channel.is_open():
-            status = self.__channel.pop()
-            if status:
-                new_time = time.time()
-                self.__log_progress(status, last_size, last_time)
-                last_time = new_time
-                last_size = status.resumable_progress
-            # Limits logs to 1 per second at most
-            time.sleep(1)
-
-    def __log_progress(self, status, previous_size, previous_time):
-        """Print log from status."""
-        progress = int(status.progress() * 100)
-        total_size = Utils.size_to_human_readable(status.total_size)
-        current_size = status.resumable_progress
-        size = Utils.size_to_human_readable(current_size)
-
-        current_time = time.time()
-        timer = time.strftime("%H:%M:%S", time.gmtime(current_time - self.__start_time))
-
-        diff_size = current_size - previous_size
-        diff_time = current_time - previous_time
-        speed = Utils.size_to_human_readable(diff_size / diff_time)
-
-        estimated_time = (status.total_size - current_size) / (diff_size / diff_time)
-        eta = time.strftime("%H:%M:%S", time.gmtime(estimated_time))
-
-        print("%s %d%% - %s/%s - %s/s - %s - ETA: %s             " %
-              (self.__operation, progress, size, total_size, speed, timer, eta), end='\r')
-
-
-class Utils:
-
-    @staticmethod
-    def describe_file(metadata):
-        """Print file metadata"""
-        print("Name: %s" % metadata["name"])
-        print("Owner: %s" % metadata["owners"][0]["displayName"])
-        if 'size' in metadata:
-            print("Size: %s" % Utils.size_to_human_readable(float(metadata["size"])))
-        if 'modifiedByMeTime' in metadata:
-            print("Modified by Me: %s" % metadata["modifiedByMeTime"])
-        else:
-            print("Modified: %s" % metadata["modifiedTime"])
-        print("ID: %s\n" % metadata["id"])
-
-    @staticmethod
-    def size_to_human_readable(num, suffix='B'):
-        """Converts bytes to a more human-readable metric"""
-        for unit in ['', 'K', 'M', 'G', 'T']:
-            if abs(num) < 1024.0:
-                return "%3.1f%s%s" % (num, unit, suffix)
-            num /= 1024.0
-        return None
-
-
-def main():
-    CommandParser().execute_command()
 
 if __name__ == '__main__':
-    main()
+    try:
+        asyncio.run(main())
```

Comment:
```
  File "./gdrive", line 16, in <module>
    asyncio.run(main())
AttributeError: module 'asyncio' has no attribute 'run'
```
Had to update to python 3.7 to be able to run it  
There are alternatives that supposedly work the same, such as  
`asyncio.get_event_loop().run_until_complete(main())`

Reply:
```python
asyncio.get_event_loop().run_until_complete(main())
```

is not exactly the same: basically we would need to copy-paste the implementation of `run()` (which uses `run_until_complete`, but handles exceptions, shutdown, etc)

@jrcamelo what do you think about requiring Python 3.9? There are other APIs I intend on using... (`asyncio.to_thread`). It was introduced in 3.9.

Reply:
Fixed retro-compatibility with 3.6. Please retest.